src/panels/lovelace/common/find-entites.ts

```diff
@@ -1,59 +1,71 @@
-import {
-  computeUnusedEntities,
-  computeUsedEntities,
-} from "./compute-unused-entities";
 import { HomeAssistant } from "../../../types";
-import { LovelaceConfig } from "../../../data/lovelace";
 import { computeDomain } from "../../../common/entity/compute_domain";
 import { HassEntity } from "home-assistant-js-websocket";
 
+const arrayFilter = (
+  array: any[],
+  conditions: Array<(value: any) => boolean>,
+  maxSize: number
+) => {
+  if (!maxSize || maxSize > array.length) {
+    maxSize = array.length;
+  }
+
+  const filteredArray: any[] = [];
+
+  for (let i = 0; i < array.length && filteredArray.length < maxSize; i++) {
+    let meetsConditions = true;
+    conditions.forEach((condition) => {
```

Comment:
Don't use forEach, it will still check the other conditions. Use `for of` with a `break`